vault/identity_store_oidc.go

```diff
@@ -310,22 +366,18 @@ func (i *IdentityStore) pathOIDCUpdateConfig(ctx context.Context, req *logical.R
 		return nil, err
 	}
 
-	var resp logical.Response
-
-	if c.Issuer != "" {
-		resp.AddWarning(`If "issuer" is set explicitly, all tokens must be ` +
-			`validated against that address, including those issued by secondary ` +
-			`clusters. Setting issuer to "" will restore the default behavior of ` +
-			`using the cluster's api_addr as the issuer.`)
-	}
-
-	i.oidcCache.Flush()
+	i.oidcCache.Flush(ns)
 
-	return &resp, nil
+	return resp, nil
 }
 
 func (i *IdentityStore) getOIDCConfig(ctx context.Context, s logical.Storage) (*oidcConfig, error) {
-	if v, ok := i.oidcCache.Get("config"); ok {
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if v, ok := i.oidcCache.Get(ns, "config"); ok {
 		return v.(*oidcConfig), nil
 	}
 
```

Comment:
Should we grab a RLock here if we're getting the config from storage?

Reply:
I think this is OK since the storage read itself is consistent (and independent of other storage ops), and we're writing the result into a concurrency-safe cache.